axon-server-connector/src/main/java/org/axonframework/axonserver/connector/heartbeat/HeartbeatConnectionCheck.java

```diff
@@ -0,0 +1,89 @@
+package org.axonframework.axonserver.connector.heartbeat;
+
+import org.axonframework.axonserver.connector.AxonServerConnectionManager;
+
+import java.time.Clock;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Consumer;
+
+import static io.axoniq.axonserver.grpc.control.PlatformOutboundInstruction.RequestCase.HEARTBEAT;
+
+/**
+ * Implementation of {@link ConnectionSanityCheck} which verifies that heartbeat is properly received.
```

Comment:
Nit: I'd either rephrase to `...that *a* heartbeat is properly...` or `...that heartbeat*s are* properly...`.